### PR TITLE
refactor: API refactor in user, organization, auth and agent module

### PR DESCRIPTION
**Pull request by KulkarniShashank**
### What ###

1. Refactored the API endpoints in the user, organization, and agent modules.
2. Improved code organization and readability.
3. Addressed code duplication and inefficiencies.
4. Updated API documentation.

### Why ###

1. To enhance the maintainability of the codebase by reducing complexity and improving structure.
2. To eliminate code redundancy and ensure consistency across the modules.
3. To provide better support for future feature development.
4. To make the API more user-friendly and intuitive for developers.

### How ###

1. Reviewed the existing codebase in the user, organization, and agent modules.
2. Identified areas for improvement, including code duplication and inconsistent naming.
3. Refactored the API endpoints using best practices and design patterns.
4. Updated API documentation to reflect the changes made.
5. Ran tests to ensure that the refactored code functions correctly.

### CODE REVIEW

File: apps/api-gateway/src/user/user.controller.ts [hunk 1]
```diff
@@ -319,96 +214,105 @@ export class UserController {
...
-
-      userDetails = await this.userService.addUserDetailsInKeyCloak(email, userInfo);
-      finalResponse = {
-        statusCode: HttpStatus.CREATED,
-        message: ResponseMessages.user.success.create,
-        data: userDetails.response
-      };
-    }
-    return res.status(HttpStatus.OK).json(finalResponse);
+  @Get('/orgs/:orgId/users')
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/user/user.controller.ts:223]: Please move this API to the `orgs` controller

File: apps/api-gateway/src/user/user.controller.ts [hunk 2]
```diff
@@ -222,37 +122,58 @@ export class UserController {
...
+      data: userDetails.response
     };
 
     return res.status(HttpStatus.OK).json(finalResponse);
-
   }
 
-  @Get('invitations')
+
+  @Get('/users/org-invitations')
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/user/user.controller.ts:148]: If `/users` is repeating for all the APIs. you can set it to the top inside `@Controllers('users')`

**Review by nishad-ayanworks — CHANGES_REQUESTED** [review]
Please resolve the comments

File: apps/api-gateway/common/exception-handler.ts [hunk 3]
```diff
@@ -15,12 +15,27 @@ export class CustomExceptionFilter extends BaseExceptionFilter {
...
-    };
+    
+    let errorResponse;
+    if (exception && exception["statusCode"] === HttpStatus.INTERNAL_SERVER_ERROR) {
+      errorResponse = {
+        statusCode: status,
+        message: 'Oops! Something went wrong. Please try again',
+        error: 'Oops! Something went wrong. Please try again'
+      };
+    } else if (exception["statusCode"] === undefined && status === HttpStatus.INTERNAL_SERVER_ERROR) {
```
Comment (nishad-ayanworks) [inline apps/api-gateway/common/exception-handler.ts:26]: Please check if `exception` is available on line 26

File: apps/api-gateway/src/authz/authz.service.ts [hunk 4]
```diff
@@ -25,4 +28,24 @@ export class AuthzService extends BaseService {
...
+    return this.sendNats(this.authServiceProxy, 'user-email-verification', payload);
+
+  }
+
+  async login(email: string, password?: string, isPasskey = false): Promise<{ response: object }> {
+    const payload = { email, password, isPasskey };
+    return this.sendNats(this.authServiceProxy, 'user-holder-login', payload);
+  }
+
+  async addUserDetailsInKeyCloak(userInfo: AddUserDetails): Promise<{ response: string }> {
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/authz/authz.service.ts:47]: Please remove this function if we are not using

File: apps/api-gateway/src/organization/organization.controller.ts [hunk 5]
```diff
@@ -39,49 +41,16 @@ export class OrganizationController {
...
-
   /**
  * 
  * @param user 
  * @param orgId 
  * @param res 
  * @returns Users list of organization
  */
-  @Get('/public')
+  @Get('/public-profile')
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/organization/organization.controller.ts:51]: please update API endpoint with `/public-profiles`

File: apps/api-gateway/src/organization/organization.controller.ts [hunk 6]
```diff
@@ -154,33 +100,66 @@ export class OrganizationController {
...
-    return res.status(HttpStatus.CREATED).json(finalResponse);
+    return res.status(HttpStatus.OK).json(finalResponse);
 
   }
 
-  @Get('/invitations/:id')
+  @Get('/:orgId/invitations')
   @ApiOperation({ summary: 'Get an invitations', description: 'Get an invitations' })
   @ApiResponse({ status: 200, description: 'Success', type: ApiResponseDto })
   @UseGuards(AuthGuard('jwt'))
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/organization/organization.controller.ts:144]: Please include `OrgRoleGuard` in the @UseGuards

File: apps/api-gateway/src/organization/organization.controller.ts [hunk 7]
```diff
@@ -193,73 +172,169 @@ export class OrganizationController {
 
   }
 
-  @Get('/dashboard')
+  @Get('/')
+  @ApiOperation({ summary: 'Get all organizations', description: 'Get all organizations' })
+  @ApiResponse({ status: 200, description: 'Success', type: ApiResponseDto })
+  @UseGuards(AuthGuard('jwt'))
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/organization/organization.controller.ts:178]: Include `OrgRoleGuard`

File: apps/api-gateway/src/user/user.controller.ts [hunk 8]
```diff
@@ -222,37 +122,58 @@ export class UserController {
...
-    required: false
+    summary: 'organization invitations',
+    description: 'Fetch organization invitations'
   })
-  async getPublicProfile(@User() reqUser: user, @Query('id') id: number, @Res() res: Response): Promise<object> {
-    const userData = await this.userService.getPublicProfile(id);
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth()
+  @ApiQuery({ name: 'limit', required: true })
+  @Roles(OrgRoles.OWNER, OrgRoles.SUPER_ADMIN, OrgRoles.ADMIN, OrgRoles.ISSUER, OrgRoles.VERIFIER, OrgRoles.MEMBER)
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/user/user.controller.ts:133]: Please remove the `Roles` and `OrgRoleGuard` from the user activity API. It is user specific API

File: apps/api-gateway/src/user/user.controller.ts [hunk 9]
```diff
@@ -222,37 +122,58 @@ export class UserController {
...
+  async getUserActivities(@Query('limit') limit: number, @Res() res: Response, @User() reqUser: user): Promise<Response> {
+
+    const userDetails = await this.userService.getUserActivities(reqUser.id, limit);
 
     const finalResponse: IResponseType = {
       statusCode: HttpStatus.OK,
-      message: ResponseMessages.user.success.fetchProfile,
-      data: userData.response
+      message: 'User activities fetched successfully',
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/user/user.controller.ts:140]: Please move this message to common library

**Review by nishad-ayanworks — CHANGES_REQUESTED** [review]
Please check comments

**Review by tipusinghaw — APPROVED** [review]
LGTM

**Review by nishad-ayanworks — APPROVED** [review]
LG

**Pull request merged**